### DISCUSSION TITLE
Another icon issue - Creating .MISS files instead of using manually placed file.

### DISCUSSION THREAD

**Discussion by My-Random-Thoughts** [dani-garcia/vaultwarden#4137] [Q&A] [answered]
So a lot of my password are for internal services and therefore don't generally have favicons.

I created some icons myself and placed them in the `icon_cache` folder and removed the corresponding `.miss` file.
This worked for a while, but now Vaultwarden doesn't show the icon and creates a `.miss` file instead.

![image](https://github.com/dani-garcia/vaultwarden/assets/21334396/45d34708-5c2d-4323-8801-2e127201d039)

I am running Vaultwarden in a docker container on a Synology NAS.  I have checked the permissions and they are correct for the folder and icons (root:root).

Is there something I can do to get the icons working??  Thanks

**Comment by BlackDex** [comment]
The icons are passed the caching time. Which means Vaultwarden checks for a fresh new version.

You could run `touch*.png` and remove all misses.

**Comment by My-Random-Thoughts** [comment]
Ah perfect, that worked, thank you.

How long is the caching time?
I can set up a cron job to touch them all every so often.

**Reply by BlackDex** [reply, marked as answer]
You can prevent it at all if you want.
https://github.com/dani-garcia/vaultwarden/blob/34e00e1478c919f725b1f94534245a55d0f5a005/.env.template#L63..L66

**Comment by chanrray** [comment]
Hi,I copy icons .png  and placed  .miss file but not working,vault is running in docker。

**Reply by chanrray** [reply]
set Negative icon cache expiry=5 and Visit this lan website again still not show icon

**Reply by chanrray** [reply]
After restart server and client It works.
What about Lan ip password,seems vault doesn't creat icon miss file
